vault/resource_database_secret_backend_connection.go

```diff
@@ -1012,6 +1029,17 @@ func getConnectionDetailsFromResponse(d *schema.ResourceData, prefix string, res
 			result["username_template"] = v.(string)
 		}
 	}
+	if v, ok := data["auth_type"]; ok {
```

Comment:
I think instead of this function, these bits can be added to the specific functions that support these parameters. If you look at [this function here](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_database_secret_backend_connection.go#L1817), it tells us which 'get connection details' function corresponds to a particular engine.

For MySQL, we should add this code to `getMySQLConnectionDetailsFromResponse`. For Postgres, I'd suggest adding a new 'get' method, similar to how you have added one for the 'set' method [here](https://github.com/hashicorp/terraform-provider-vault/pull/2012/files#diff-411fc4f295438ae111dc09ea6e3b487b551ccb94cf5de06d56c4606f1e052c5dR1416), that calls the existing method and then gets these parameters from the response

Reply:
You could also add a `getCloudDatabaseConnectionData` to adhere to your `set` pattern, and this can then be used by both postgres and mySQL

Reply:
I wanted to do it that way, but because of the way we chain getConnectionDetails calls, it wasn't possible to do without deviating from the existing pattern. That, combined with the fact that we have a decent amount of parameter name duplication in other getConnectionDetails functions, convinced me to just add the getters in both MySQL and Postgres's special functions.

I also left a comment at the top level of this review about this.